test/unit/math/test_ad.hpp

```diff
@@ -1724,6 +1774,75 @@ void expect_ad_matvar_v(const ad_tolerances& tols, const F& f,
   test_matvar_gradient(tols, A_mv_f, A_vm_f, A_mv, A_vm);
 }
 
+/**
+ * For an unary function check that an Eigen matrix of vars and a var with an
+ * inner eigen type return the same values and adjoints to within
+ * the given tolerances. This is done by calling the function on both types,
+ * summing the results, then taking the gradient of the sum which will
+ * propogate the adjoints upwards.
+ *
+ * @tparam F A lambda or functor type that calls the unary function.
+ * @tparam T An Eigen matrix type
+ * @param tols tolerances for test
+ * @param f a lambda
+ * @param x An Eigen matrix.
+ */
+template <typename F, typename EigMat, require_eigen_t<EigMat>* = nullptr>
+void expect_ad_matvar_v(const ad_tolerances& tols, const F& f,
+                        const std::vector<EigMat>& x) {
+  using stan::plain_type_t;
+  using stan::math::promote_scalar_t;
+  using stan::math::var;
+  using stan::math::var_value;
+  using stan::math::test::type_name;
+  using vec_mat_var = std::vector<promote_scalar_t<var, EigMat>>;
+  using vec_var_mat = std::vector<var_value<plain_type_t<EigMat>>>;
+  vec_mat_var A_mv;
```

Comment:
I think it the future it would be good to change `_mv` to `_mat_var` etc. but for now let's keep them all in the same style